### PR TITLE
Replace blacklist/whitelist with include/exclude in config (fixes #1154)

### LINKED ISSUE #1154: Replace terms blacklist & whitelist in configuration
@simonihmig and myself had a quick chat on discord about the configuration options `blacklist` and `whitelist`. We decided to rename them. Our feature is similar to the one currently called `blacklist` and `whitelist` in Ember itself. [RFC 639](https://github.com/emberjs/rfcs/pull/639) proposed to rename them. The RFC is currently in final comment period. We decided to follow the chosen naming in that RFC as soon as it's merged. The old names should be deprecated but not removed yet.

### PR DESCRIPTION
**Pull request by seanCodes**
Use more inclusive language by replacing `blacklist` and `whitelist` in the configuration options with `exclude` and `include` (as was decided in #1154).

> **Warning**: This a **breaking change** currently. I noticed a new major version is in the works so I didn’t bother to make it backwards compatible, but this could easily be done if that’s preferred.

Also in this PR (but unrelated) is a change to the `.prettierignore` to add `*.hbs` since my editor (VS Code) kept trying to prettify hbs files on save.

### CODE REVIEW

**Review by jelhan — APPROVED** [review]
Looks good to me. Thanks a lot!

If you have time, adding an build-time error if `whitelist` or `blacklist` options are set, would be great. I fear otherwise people may easily miss this breaking change.

**Comment by jelhan** [comment]
@seanCodes please have a look at this two failing tests:

```
  1) index
       generateExcludeList
         `include` list is correct for simple component:
     TypeError: addonIndex.generateExcludeList is not a function
      at Context.<anonymous> (node-tests/index-tests.js:56:31)
      at processImmediate (node:internal/timers:466:21)

  2) index
       generateExcludeList
         `include` list is correct for component with dependencies:
     TypeError: addonIndex.generateExcludeList is not a function
      at Context.<anonymous> (node-tests/index-tests.js:62:31)
      at processImmediate (node:internal/timers:466:21)
```

The other failing tests are not related to this PR. We should fix them in a separate PR before merging this one. Will try finding some time doing so.

**Comment by seanCodes** [comment]
@jelhan Thanks for reviewing! I’ll add the errors and look into those tests.

**Comment by seanCodes** [comment]
@jelhan This should be ready for re-review. 👍🏽  The tests were fixed and errors were added for the old options.

**Comment by seanCodes** [comment]
@jelhan Bump here, anything blocking this?

File: .prettierignore [hunk 1]
```diff
@@ -1,3 +1,5 @@
+*.hbs
```
Comment (jelhan) [inline .prettierignore:1]: I feel we should implement #1432 soonish. I also struggled with IDE being setup to run Prettier and seeing templates being changed. Configuring Prettier to ignore Glimmer templates seems to be a good interim solution. 👍

**Review by jelhan — APPROVED** [review]
Thanks a lot. Looks good to me. I'm sorry that review took that long.

**Comment by jelhan** [comment]
Reopened to trigger CI. Somehow it wasn't run yet. Or results weren't available anymore.

**Pull request merged**